Author email optional

brbrbr opened the pull request:
Made the emails to the 'created by' user optional
Added modified by as option to send email to
Tweaked the limit to 5
Made a true administrator edit link
Changed the xhtml setting. Not sure about this. It confused at least thunderbird and gmail

brbrbr committed:
Made the emails to the 'created by' user optional
Added modified by as option to send email to
Tweaked the limit to 5
Made a true administrator edit link
Changed the xhtml setting. Not sure about this. It confused at least thunderbird and gmail

brbrbr committed:
Overlooked onBuildAdministratorLoginURL

zero-24 committed:
Merge branch 'master' into skip-author_email

zero-24 commented on forms/sendForm.xml:
```diff
@@ -94,6 +94,18 @@
+				<field
+				name="who_email"
+				type="list"
+				label="PLG_TASK_REVIEWCONTENTNOTIFICATION__WHO_EMAIL_LBL"
```
```suggestion
				label="PLG_TASK_REVIEWCONTENTNOTIFICATION_WHO_EMAIL_LABEL"
```

zero-24 commented on language/en-GB/plg_task_reviewcontentnotification.ini:
```diff
@@ -9,6 +9,11 @@ PLG_TASK_REVIEWCONTENTNOTIFICATION_DATE_MODIFIER_TYPE_MONTHS="Months"
 PLG_TASK_REVIEWCONTENTNOTIFICATION_DATE_MODIFIER_TYPE_YEARS="Years"
 PLG_TASK_REVIEWCONTENTNOTIFICATION_EMAIL_DESC="A comma separated list of the email addresses which will receive all notification emails. If it's left blank, and the author is not set all Super Users of this site will receive the notification email."
 PLG_TASK_REVIEWCONTENTNOTIFICATION_EMAIL_LBL="Special eMails"
+PLG_TASK_REVIEWCONTENTNOTIFICATION__WHO_EMAIL_LBL="Send email to"
```
```suggestion
PLG_TASK_REVIEWCONTENTNOTIFICATION_WHO_EMAIL_LABEL="Send email to"
```

zero-24 commented on src/Extension/ReviewContentNotification.php:
```diff
@@ -576,20 +590,36 @@ private function getSecondNotificationDateByArticleId($articleId)
         }
 
+        if ($whoEmail == 'none') {
+            if (\count($recipients) && $whoEmail == 'none') {
```
```suggestion
            if (\count($recipients)) {
```

whoEmail is checked above right?

brbrbr replied:
I have to rework this part. I overlooked the created_by in the loadUserById further below in the code.

zero-24 replied:
Ok let me know once done.

brbrbr replied:
Ok,
I made the 'Send email to' optional so you can pick which users to send.

I made some changes to prevent duplicate emails and validated the result of loadUserById.


Just for the background; For project with a lot of categories and articles, we want to send emails to one fixed address. Hence this PR and #5.

We also want to send an aggregated list instead of single emails. I solved that issue as wel. But it's a kind of ugly. I can make a PR for that if you like to

zero-24 commented on forms/sendForm.xml:
```diff
@@ -79,11 +79,11 @@
-				min="10"
-				step="10"
+				min="5"
+				step="5"
```
```suggestion
				step="1"
```

zero-24 commented on forms/sendForm.xml:
```diff
@@ -94,6 +94,18 @@
 				description="PLG_TASK_REVIEWCONTENTNOTIFICATION_EMAIL_DESC"
 				default=""
 			/>
+				<field
```
```suggestion
			<field
```

zero-24 reviewed (changes requested):
Thanks for the PR I have commented here some suggestions. Please take a look into it :)

brbrbr committed:
Update forms/sendForm.xml

Co-authored-by: Tobias Zulauf <zero-24@users.noreply.github.com>

brbrbr committed:
Update language/en-GB/plg_task_reviewcontentnotification.ini

Co-authored-by: Tobias Zulauf <zero-24@users.noreply.github.com>

brbrbr committed:
Update forms/sendForm.xml

Co-authored-by: Tobias Zulauf <zero-24@users.noreply.github.com>

brbrbr committed:
Update forms/sendForm.xml

Co-authored-by: Tobias Zulauf <zero-24@users.noreply.github.com>

brbrbr committed:
Reworked Send email to

brbrbr committed:
Reworked Send email to

brbrbr committed:
Reworked Send email to

brbrbr committed:
liked the list layout better
and an empty list is not stored in the params so $event->getArgument('params')->who_email returns null where it should be [] and the null-coalescing would work.
fixed default value.
 Please enter the commit message for your changes. Lines starting

zero-24 commented on forms/sendForm.xml:
```diff
@@ -97,12 +97,13 @@
 				label="PLG_TASK_REVIEWCONTENTNOTIFICATION_WHO_EMAIL_LABEL"
 				description="PLG_TASK_REVIEWCONTENTNOTIFICATION_WHO_EMAIL_DESC"
-				checkedOptions="created"
+				default='["created"]'
```
```suggestion
				default="created"
```

zero-24 commented on src/Extension/ReviewContentNotification.php:
```diff
@@ -99,9 +99,9 @@ private function checkReviewContentNotification(ExecuteTaskEvent $event): int
+        $whoEmail               = $event->getArgument('params')->who_email ?? [];
         $forcedLanguage         = $event->getArgument('params')->language_override ?? 'user';
-
+      
```
```suggestion

```

brbrbr replied:
The default value in '$event->getArgument('params')->who_email ?? [];' introduces a backward compatibility. The list field will be null for 'never selected (new)' as well as for 'saved but nothing selected'. So I can't use a default value of ['created'] which would be backward compatible. As a result, the task will fall back to the superusers.

zero-24 reviewed (changes requested):
small changes

zero-24 commented on src/Extension/ReviewContentNotification.php:
```diff
@@ -120,26 +121,8 @@ private function checkReviewContentNotification(ExecuteTaskEvent $event): int
-         */
-        $this->getApplication()->triggerEvent('onBuildAdministratorLoginURL', [&$backendURL]);
+
+
```
```suggestion
```

zero-24 commented on src/Extension/ReviewContentNotification.php:
```diff
@@ -576,54 +589,71 @@ private function getSecondNotificationDateByArticleId($articleId)
             }
         }
+        $users = [];
+        if (in_array('created', $whoEmail)) {
```
```suggestion

        if (in_array('created', $whoEmail)) {
```

zero-24 commented on src/Extension/ReviewContentNotification.php:
```diff
@@ -576,54 +589,71 @@ private function getSecondNotificationDateByArticleId($articleId)
     {
         $recipients = [];
 
+        //prepare  the value of forcedLanguage for future use. 
```
```suggestion
        // Prepare the value of forcedLanguage for future use. 
```

zero-24 commented on src/Extension/ReviewContentNotification.php:
```diff
@@ -576,54 +589,71 @@ private function getSecondNotificationDateByArticleId($articleId)
         $recipients = [];
 
+        //prepare  the value of forcedLanguage for future use. 
+        //forcedLanguage is used as a 'boolean' as well as value
```
```suggestion
        // forcedLanguage is used as a 'boolean' as well as value
```

zero-24 commented on src/Extension/ReviewContentNotification.php:
```diff
@@ -576,54 +589,71 @@ private function getSecondNotificationDateByArticleId($articleId)
+                    $recipients[$superUser->email] = ['email' => $superUser->email, 'language' => $language];
                 } else {
-                    $recipients[] = ['email' => $superUser->email, 'language' => empty($forcedLanguage) ? $currentSiteLanguage : $forcedLanguage];
+                    //this avoid duplicates. 
```
```suggestion
                    // This avoid duplicates. 
```

brbrbr committed:
default from multiple  to single notation

Co-authored-by: Tobias Zulauf <zero-24@users.noreply.github.com>

brbrbr committed:
Code style

Co-authored-by: Tobias Zulauf <zero-24@users.noreply.github.com>

brbrbr committed:
Code style

Co-authored-by: Tobias Zulauf <zero-24@users.noreply.github.com>

brbrbr committed:
code style

Co-authored-by: Tobias Zulauf <zero-24@users.noreply.github.com>

brbrbr committed:
code style

Co-authored-by: Tobias Zulauf <zero-24@users.noreply.github.com>

brbrbr committed:
code style

Co-authored-by: Tobias Zulauf <zero-24@users.noreply.github.com>

brbrbr committed:
code style

Co-authored-by: Tobias Zulauf <zero-24@users.noreply.github.com>

zero-24 commented:
Merging this here now, will next have a look into #5. Thank you :)
